rmgpy/rmg/react.py

```diff
@@ -118,59 +117,70 @@ def react_all(core_spc_list, numOldCoreSpecies, unimolecularReact, bimolecularRe
         a list of lists of reactions generated from each species tuple
         a list of species tuples corresponding to each list of reactions
     """
+    family_names, families = zip(*getDB('kinetics').families.items())
+
+    # List of families that should not react together as they are likely to generate a lot of reactions and
+    # therefore negatively impact load balancing for multiprocessing
+    major_families = [
+        'H_Abstraction', 'R_Recombination', 'Intra_Disproportionation', 'Intra_RH_Add_Endocyclic',
+        'Singlet_Carbene_Intra_Disproportionation', 'Intra_ene_reaction', 'Disproportionation',
+        '1,4_Linear_birad_scission', 'R_Addition_MultipleBond', '2+2_cycloaddition_Cd', 'Diels_alder_addition',
+        'Intra_RH_Add_Exocyclic', 'Intra_Retro_Diels_alder_bicyclic', 'Intra_2+2_cycloaddition_Cd',
+        'Birad_recombination', 'Intra_Diels_alder_monocyclic', '1,4_Cyclic_birad_scission', '1,2_Insertion_carbene',
+    ]
+
+    # Only employ family splitting for reactants that have a larger number than min_atoms
+    min_atoms = 10
+
     # Select reactive species that can undergo unimolecular reactions:
-    spc_tuples = [(core_spc_list[i],)
-                  for i in xrange(numOldCoreSpecies) if (unimolecularReact[i] and core_spc_list[i].reactive)]
+    spc_tuples = []
+    for i in xrange(numOldCoreSpecies):
+        fam_leftovers = []
+        for k, family in enumerate(family_names):
+            # Find reactions involving the species that are unimolecular and only generate tuple if family is not
+            # bimolecular in forward and backward direction.
+            if not (len(families[k].forwardTemplate.reactants)==2 and len(families[k].forwardTemplate.products)==2):
```

Comment:
Probably doesn't make a difference performance-wise, but seems more intuitive to check `reactants == 1 or products == 1`.

Similarly, for bimolecular, you could check `reactants == 2 or products == 2`.